### PR TITLE
Standardize #if defined constructs

### PR DESCRIPTION
**Pull request by richsalz**
Always use "#if defined(xxx)" not "defined (xxx)" or "defined xxx"

This might help (I hope) in removing some old platforms.

### CODE REVIEW

File: crypto/o_dir.c [hunk 1]
```diff
@@ -23,14 +23,14 @@
 #include "internal/o_dir.h"
 
 #define LPDIR_H
-#if defined OPENSSL_SYS_UNIX || defined DJGPP \
-    || (defined __VMS_VER && __VMS_VER >= 70000000)
+#if defined OPENSSL_SYS_UNIX || defined(DJGPP) \
```
Comment (botovq) [inline crypto/o_dir.c:26]:
```suggestion
#if defined(OPENSSL_SYS_UNIX) || defined(DJGPP) \
```

(this will conflict with the DJGPP PR)

**Review by botovq — COMMENTED** [review]
Consistency is a good thing. There's still a lot of `ifdef` and `ifndef` left, so I don't know how much that will help. The `unifdef` tool (ported to many plaftorms) is tremendously helpful for such grunt work.

**Comment by richsalz** [comment]
There are over 4000 each of ifdef and ifndef, ugh.  Does ifndef handle the multiple conditionals properly?

**Comment by botovq** [comment]
It can handle them but it does need a lot of hand holding especially in an ifdef mess like this code base.

**Comment by richsalz** [comment]
I did `brew install ifdef` and am playing with it.  It's gotten much better since I posted to Usenet in 1986 ;) https://chambers.nycbug.org/pub/The_Unix_Archive/Unix_Usenet/comp.sources.unix/1986-July/005314.html

**Pull request closed without merging**